openverse_catalog/dags/providers/provider_api_scripts/finnish_museums.py

```diff
@@ -42,54 +42,153 @@ class FinnishMuseumsDataIngester(ProviderDataIngester):
     batch_limit = 100
     delay = 5
     format_type = "0/Image/"
+    buildings = [
+        "0/Suomen kansallismuseo/",
+        "0/Museovirasto/",
+        "0/SATMUSEO/",
+        "0/SA-kuva/",
+    ]
 
     def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        # A flag that is turned on each time we start ingestion for a new set of
-        # query params. This is useful for logging information on only the first
-        # batch of each set.
+        """
+        This DAG runs ingestion separately for each configured `building`. When a
+        building has many records for the ingestion date, the DAG further splits up
+        ingestion into time slices. Each run of the `ingest_records` function for
+        a particular (building, time slice) pair is an "iteration" of the DAG.
+
+        For logging and alerting purposes, we maintain several instance variables
+        that help track each iteration.
+        """
+        # A flag that is True only when we are processing the first batch of data in
+        # a new iteration.
         self.new_iteration = True
+        # Use to keep track of the number of records we've fetched from the API so far,
+        # specifically in this iteration.
+        self.fetched_count = 0
 
```

Comment:
Pycharm suggests adding `timestamp_pairs` here in the `__init__()`, given it's used later in the `ingest_records` method.
```suggestion
        self.fetched_count = 0
	self.timestamp_pairs = None
	
```

Reply:
Thank you! Now you point it out, `timestamp_pairs` can actually just be a local variable to `ingest_records`.